File: python/Etc/diagonal_difference.py
# https://www.hackerrank.com/challenges/diagonal-difference/problem

def diagonalDifference(arr):
    column_idx = 0
    left_to_right = 0
    right_to_left = 0
    for row in arr:
        left_to_right += row[column_idx]
        right_to_left += row[-1-column_idx]
        column_idx += 1
    
    return abs(left_to_right - right_to_left)

File: go/hackerrank/diagonalDifference.go
// https://www.hackerrank.com/challenges/diagonal-difference/problem

func diagonalDifference(arr [][]int32) int32 {
    primaryDiagonal := 0
    secondaryDiagonal := 0
    for row, name := range(arr) {
        for column, num := range(name) {
            if (row == column){
                primaryDiagonal += int(num)
            }
            if (column == len(arr) - row - 1) {
                secondaryDiagonal += int(num)
            }
        }
    }

    if(primaryDiagonal >= secondaryDiagonal){
        return int32(primaryDiagonal - secondaryDiagonal)
    }

    return int32(secondaryDiagonal - primaryDiagonal)
}
